libwallet/assets_manager.go

```diff
@@ -923,12 +924,13 @@ func (mgr *AssetsManager) InitializeDEX(ctx context.Context) {
 		return
 	}
 
-	if mgr.DexcInitialized() || mgr.startingDEX.Load() {
+	// Prevent multiple initialization.
+	if mgr.DEXCInitialized() || mgr.startingDEX.Load() {
```

Comment:
Use `CompareAndSwap(false, true)` instead of `Load` to immediately block multiple initialization.